docs/rules/no-type-literal.md

```diff
@@ -0,0 +1,121 @@
+# Enforces the use of interfaces over type literals (`type T = {...}`).
```

Comment:
(Sorry not on computer right now, so can't do much digging)

We have the "allowUnion" and "allowIntersection" configuration in this rule which broadens it from just a literal, right? Plus should we not actually be including your callback example in this rule as well?

e.g. should we not be preferring `func2` here?

https://www.typescriptlang.org/play/index.html#src=type%20func%20%3D%20()%20%3D%3E%20true%0D%0A%0D%0Ainterface%20func2%20%7B%0D%0A%20%20%20%20()%3A%20true%3B%0D%0A%7D%0D%0A%0D%0Aconst%20test%3A%20func%20%3D%20()%20%3D%3E%20true%0D%0Aconst%20test2%3A%20func2%20%3D%20()%20%3D%3E%20true